data-exporter/src/main/java/org/reactome/release/dataexport/UniProtReactomeEntry.java

```diff
@@ -303,6 +373,37 @@ private void setDisplayName(String displayName) {
 		this.displayName = displayName;
 	}
 
+	/**
+	 * Checks if a passed String is a valid canonical UniProt accession (i.e. 6 or 10 alphanumeric characters matching
+	 * the regular expression "[OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z][0-9]([A-Z][A-Z0-9]{2}[0-9]){1,2}" as defined at
+	 * https://www.uniprot.org/help/accession_numbers as of October 2019).
+	 * E.g. P12345
+	 * @param accession UniProt accession to test for validity
+	 * @return true if the accession passed is a valid canonical (i.e. not an isoform) UniProt accession,
+	 * false otherwise
+	 */
+	private boolean isValidCanonicalAccession(String accession) {
+		final Pattern ANCHORED_PARENT_ACCESSION_REGEX = Pattern.compile("^" + PARENT_ACCESSION_REGEX.pattern() + "$");
```

Comment:
I am confused about why it is `final`, and why it is placed here in this function instead of at the top of the class?

Reply:
Some developers declare everything final, except for what they _know_ will need to be mutable. It's not necessary, it's just a habit.

Reply:
...although I agree that in a two line function, declaring the _only_ variable as `final` seems unnecessary. But it doesn't do any harm.

Reply:
I usually prefer to make my constants final even if it's scoped to a method.  @SolomonShorser-OICR is right - it's not necessary but just an indication I use to show it's meant to give a name to something constant and cryptic/magical.